applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_interface_stress_state_policy.cpp

```diff
@@ -0,0 +1,148 @@
+
+    Vector shape_function_values(2);
+    shape_function_values <<= 1.0, 2.0;
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        [[maybe_unused]] const auto b_matrix =
+            stress_state_policy.CalculateBMatrix({}, shape_function_values, geometry),
+        "Each node pair must have exactly one shape function value, cannot compute the B matrix.");
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceStressState_ReturnsExpectedVoigtSize, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto stress_state_policy = InterfaceStressState{};
+
+    KRATOS_EXPECT_EQ(stress_state_policy.GetVoigtSize(), 2);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceStressState_ReturnsExpectedVoigtVector, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto stress_state_policy = InterfaceStressState{};
+
+    const auto& voigt_vector = stress_state_policy.GetVoigtVector();
+
+    Vector expected_voigt_vector(2);
+    expected_voigt_vector <<= 1.0, 0.0;
+    KRATOS_EXPECT_VECTOR_NEAR(expected_voigt_vector, voigt_vector, 1.0e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceStressState_ReturnsCorrectBMatrixForThreePlusThreeNodesGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto stress_state_policy = InterfaceStressState{};
+    const auto geometry            = CreateThreePlusThree2DLineInterfaceGeometry();
+
+    Vector shape_function_values(3);
+    shape_function_values <<= -0.125, 0.375, 0.75; // Shape function values for xi = 0.5
+
+    const auto b_matrix = stress_state_policy.CalculateBMatrix({}, shape_function_values, geometry);
+
+    // clang-format off
+    Matrix expected_b_matrix(2, 12);
+    expected_b_matrix <<= 0,   0.125, 0, -0.375, 0, -0.75, 0, -0.125, 0, 0.375, 0, 0.75,
+                          0.125, 0, -0.375, 0, -0.75, 0, -0.125, 0, 0.375, 0, 0.75, 0;
+    // clang-format on
+
+    KRATOS_EXPECT_MATRIX_NEAR(b_matrix, expected_b_matrix, 1.0e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceStressState_ReturnsCorrectIntegrationCoefficient, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto interface_stress_state = InterfaceStressState{};
+
+    const Geometry<Node>::IntegrationPointType integration_point(0.5, 0.3, 0.0, 0.5);
+
+    constexpr auto detJ                   = 2.0;
+    const auto     calculated_coefficient = interface_stress_state.CalculateIntegrationCoefficient(
+        integration_point, detJ, LineInterfaceGeometry<Line2D3<Node>>());
+
+    // The expected number is calculated as follows:
+    // weight * detJ = 0.5 * 2.0 = 1.0
+    KRATOS_EXPECT_NEAR(calculated_coefficient, 1.0, 1e-5);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceStressState_Throws_WhenAskingForStrain, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto stress_state_policy = InterfaceStressState{};
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        [[maybe_unused]] const auto strain = stress_state_policy.CalculateGreenLagrangeStrain({}),
+        "For interfaces, it is not possible to calculate the Green "
```

Comment:
I think that mostly the name is written with a - as Green-Lagrange strain Preferably that reflects in the error message.